src/sentry/static/sentry/app/views/alerts/rules/filter.tsx

```diff
@@ -0,0 +1,185 @@
+    const checkedQuantity = this.getNumberOfActiveFilters();
+
+    const dropDownButtonProps: Pick<DropdownButtonProps, 'children' | 'priority'> & {
+      hasDarkBorderBottomColor: boolean;
+    } = {
+      children: (
+        <React.Fragment>
+          <IconFilter size="xs" />
+          <FilterLabel>{t('Filter')}</FilterLabel>
+        </React.Fragment>
+      ),
+      priority: 'default',
+      hasDarkBorderBottomColor: false,
+    };
+
+    if (checkedQuantity > 0) {
+      dropDownButtonProps.children = (
+        <span>{tn('%s Active Filter', '%s Active Filters', checkedQuantity)}</span>
+      );
+      dropDownButtonProps.priority = 'primary';
+      dropDownButtonProps.hasDarkBorderBottomColor = true;
+    }
+    return (
+      <DropdownControl
+        menuWidth="240px"
+        blendWithActor
+        button={({isOpen, getActorProps}) => (
+          <StyledDropdownButton
+            {...getActorProps()}
+            showChevron={false}
+            isOpen={isOpen}
+            hasDarkBorderBottomColor={dropDownButtonProps.hasDarkBorderBottomColor}
+            priority={dropDownButtonProps.priority as DropdownButtonProps['priority']}
+            data-test-id="filter-button"
+          >
+            {dropDownButtonProps.children}
+          </StyledDropdownButton>
+        )}
+      >
+        <Header>
+          <span>{header}</span>
+          <CheckboxFancy
+            isChecked={checkedQuantity > 0}
+            isIndeterminate={checkedQuantity > 0 && checkedQuantity !== filterList.length}
+            onClick={event => {
+              event.stopPropagation();
+              this.toggleAllFilters();
+            }}
+          />
+        </Header>
+        {children({toggleFilter: this.toggleFilter})}
+      </DropdownControl>
+    );
+  }
+}
+
+const Header = styled('div')`
+  display: grid;
+  grid-template-columns: auto min-content;
+  grid-column-gap: ${space(1)};
+  align-items: center;
+
+  margin: 0;
+  background-color: ${p => p.theme.backgroundSecondary};
+  color: ${p => p.theme.gray300};
+  font-weight: normal;
+  font-size: ${p => p.theme.fontSizeMedium};
+  padding: ${space(1)} ${space(2)};
+  border-bottom: 1px solid ${p => p.theme.border};
+`;
+
+const FilterLabel = styled('span')`
+  margin-left: ${space(1)};
+`;
+
+const StyledDropdownButton = styled(DropdownButton)<{hasDarkBorderBottomColor?: boolean}>`
+  white-space: nowrap;
+  max-width: 200px;
+
+  z-index: ${p => p.theme.zIndex.dropdown};
+
+  &:hover,
+  &:active {
+    ${p =>
+      !p.isOpen &&
+      p.hasDarkBorderBottomColor &&
+      `
+          border-bottom-color: ${p.theme.button.primary.border};
+        `}
+  }
+
+  ${p =>
+    !p.isOpen &&
+    p.hasDarkBorderBottomColor &&
+    `
+      border-bottom-color: ${p.theme.button.primary.border};
+    `}
+`;
+
+export const List = styled('ul')`
+  list-style: none;
+  margin: 0;
+  padding: 0;
+`;
+
+export const ListItem = styled('li')<{isChecked?: boolean}>`
```

Comment:
I think either let's move this out of this file and into `rules/index.tsx` or better move rendering the items into here like in `spans/filters.tsx`